### DISCUSSION TITLE
Nothing saving (Heroku)

### DISCUSSION THREAD

**Discussion by Po-rygon** [coder/code-server#3758] [Q&A] [answered]
Hi, it seems that nothing I do, (install extensions, write files, change theme) are saving in code-server, I’m using Brave browser, Mac os, hosting on heroku. Any idea how to save files?

**Comment by jsjoeio** [comment]
Hmm that's strange. What version of code-server are you using? Is it possible for you to post a screen recording so we can see?

Does the same behavior happen in Chrome/Safari/Firefox?

**Reply by Po-rygon** [reply]
yes, Also I tried using rclone and it didn’t really help, Not sure if the screen recording is possible because it takes them about 3 hours to disappear

**Reply by jsjoeio** [reply]
That's very strange  🤔 Might be worth opening an issue under http://github.com/cdr/deploy-code-server and providing steps to reproduce. Then tag me or @bpmct and we can test it out

**Comment by bpmct** [comment, marked as answer]
Hey @BinaryEgypt.

With Heroku and many other app platforms, there is no concept of a persistent filesystem. This means anything that is not defined in the image (e.g Dockerfile and files copied from the github repo) will be lost. In Heroku's case, the app shuts down when inactive, so every 3 hours or so totally makes sense.

This is documented in "persist your filesystem with rclone": https://github.com/cdr/deploy-code-server/tree/main/deploy-container#-persist-your-filesystem-with-rclone. With rclone, you will need to "push" your changes to the remote (google drive, s3, etc) using the command or button in VS Code.

Ideally, your source code itself is saved (and pushed to) the GitHub repository. For stuff like VS Code settings and other dependencies (python, java, pip, etc), you can install this in the Dockerfile for your repo: https://github.com/cdr/deploy-code-server/tree/main/deploy-container#modifying-your-code-server-environment

If you'd rather an environment that you can "set and forget" without having to source control/push/rebuild, you may want to install code-server on a VM instead: https://github.com/cdr/deploy-code-server#code-server-on-a-vm-vs-a-container

**Reply by Po-rygon** [reply]
Thanks for the reply! Just wanted to know how I could push.
